File: bioinformatics/routes/revcomp.js
var express = require('express');
var router = express.Router();
var exec = require("child_process").exec;

/* GET home page. */
router.get('/', function(req, res, next) {
  console.log("## GET: revcomp");
  res.render('revcomp', { input: "", result: "", option: "" });
});

router.post('/', function(req, res, next) {
  console.log("## POST: revcomp");
  var result = "";
  var seq = req.body.seq;
  var option = "";
  seq = seq.toUpperCase();
  if(req.body.rev !== undefined)
    option += "rev";
  if(req.body.comp !== undefined)
    option += "comp";

  var cmd = "python public/scripts/revComp.py "+seq+" "+option;
  console.log("cmd: "+cmd);
  console.log("seq: "+seq);
  console.log("opt: "+option);
  if(seq){
    exec(cmd, function(err,stdout,stderr){
      console.log("stdout: "+stdout);
      console.log("stderr: "+stderr);
      res.render('revcomp', {input: seq, result: stdout, option: option});
    });
  }
  
  //res.render('revcomp', { result: result });
});

module.exports = router;

File: bioinformatics_ori/public/scripts/revComp.py
#!/usr/bin/python

import sys

if len(sys.argv) != 3:
    sys.exit()

def rev(seq):
    revSeq = seq[::-1]
    return revSeq

def comp(seq):
    compSeq = ""
    d = {"A":"T","C":"G","G":"C","T":"A"}
    for s in seq:
        compSeq += d[s]
    return compSeq

seq = sys.argv[1]
mode = sys.argv[2]
result = ""

if mode == "rev":
    result = rev(seq)
elif mode == "comp":
    result = comp(seq)
elif mode == "revcomp":
    result = comp(rev(seq))

print(result)


